toolkits/build.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+TRT_IMAGE_TAG=24.12-py3
+TORCH_VARIANT=cpu
+PYTHON_VERSION=3.12
+UBUNTU_VERSION=24.04
+TARGET=cpu
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+
+CUDA_TOOLKIT=0
+ML_TOOLKIT=0
+
+BUILD_FLAGS=()
+while [ "$#" -gt 0 ]; do
+  case "$1" in
+  --cuda-toolkit)
+    CUDA_TOOLKIT=1
+    shift 1
+    ;;
+  --ml-toolkit)
+    ML_TOOLKIT=1
+    shift 1
+    ;;
+  --tensorrt-image-tag)
+    TRT_IMAGE_TAG=$2
+    shift 2
+    ;;
+  --python-version)
+    PYTHON_VERSION=$2
+    shift 2
+    ;;
+  --ubuntu-version)
+    UBUNTU_VERSION=$2
+    shift 2
+    ;;
+  --torch-variant)
+    if [[ "$2" == "cpu" || "$2" == "gpu" ]]; then
+      TORCH_VARIANT=$2
+      shift 2
+    else
+      echo "Invalid torch variant specified. Use 'cpu' or 'gpu'."
+      exit 1
+    fi
+  --target)
+    if [[ "$2" == "cpu" || "$2" == "gpu" ]]; then
+      TARGET=$2
+      shift 2
+    else
+      echo "Invalid target specified. Use 'cpu' or 'gpu'."
+      exit 1
+    fi
+    ;;
+    shift 2
+    ;;
+  -r | --rebuild)
+    BUILD_FLAGS+=(--no-cache)
+    shift 1
+    ;;
+  -v | --verbose)
+    BUILD_FLAGS+=(--progress=plain)
+    shift 1
+    ;;
+  *)
+    echo "Unknown option: $1" >&2
+    exit 1
+    ;;
+  esac
+done
+
+if [ $CUDA_TOOLKIT -eq 0 ] && [ $ML_TOOLKIT -eq 0 ]; then
+  echo "No toolkit selected to build, nothing to do."
+elif [ $CUDA_TOOLKIT -eq 1 ]; then
+  VERSION=$(cat "${SCRIPT_DIR}"/VERSION.cuda)
+  IMAGE_NAME="ghcr.io/aica-technology/cuda-toolkit"
+elif [ $ML_TOOLKIT -eq 1 ]; then
+  VERSION=$(cat "${SCRIPT_DIR}"/VERSION.ml)
+  IMAGE_NAME="ghcr.io/aica-technology/ml-toolkit"
+
+  BUILD_FLAGS+=(--build-arg=UBUNTU_VERSION=${UBUNTU_VERSION})
+  BUILD_FLAGS+=(--build-arg=PYTHON_VERSION=${PYTHON_VERSION})
+  BUILD_FLAGS+=(--build-arg=TORCH_VARIANT=${TORCH_VARIANT})
+  BUILD_FLAGS+=(--target ${TARGET})
+fi
+
+BUILD_FLAGS+=(--build-arg=TRT_IMAGE_TAG=${TRT_IMAGE_TAG})
+BUILD_FLAGS+=(--build-arg=VERSION=${VERSION})
+docker buildx build -t "${IMAGE_NAME}":v"${VERSION}" "${BUILD_FLAGS[@]}" .
```

Comment:
ToDo: I'll append gpu/cpu/cu versions eventually, but feel free to share your preferred naming convention